### ISSUE TITLE
cluster-cluster is killed when member from other cluster tries to join

### ISSUE THREAD

**Issue by mariusgrigaitis** [coreos/etcd#2836]
etcd 2.1.0-alpha.0

1) Start a cluster of 3 nodes
2) Start another cluster of 1 node
3) Stop another cluster
4) Join another cluster to 1) cluster

Result:
etcd is not running on all instances (killed)

Output from another cluster:
```
2015/05/15 15:18:28 INFO: raft: ce2a822cea30bfca is starting a new election at term 2
2015/05/15 15:18:28 INFO: raft: ce2a822cea30bfca became candidate at term 3
2015/05/15 15:18:28 INFO: raft: ce2a822cea30bfca received vote from ce2a822cea30bfca at term 3
2015/05/15 15:18:28 INFO: raft: ce2a822cea30bfca became leader at term 3
2015/05/15 15:18:28 INFO: raft.node: ce2a822cea30bfca elected leader ce2a822cea30bfca at term 3
2015/05/15 15:18:28 etcdserver: published {Name:soad-cluster-node1 ClientURLs:[http://172.21.3.1:2379 http://172.21.3.1:4001]} to cluster 7e27652122e8b2ae
2015/05/15 15:18:28 rafthttp: fail to find sender 7f6b262dd7767f30
2015/05/15 15:18:28 rafthttp: fail to find sender 7f6b262dd7767f30
2015/05/15 15:18:28 rafthttp: fail to find sender 89f14c2ba6df79a3
2015/05/15 15:18:28 rafthttp: fail to find sender 89f14c2ba6df79a3
2015/05/15 15:18:28 rafthttp: fail to find sender 69e5eeba42a3e4cb
2015/05/15 15:18:28 rafthttp: fail to find sender 69e5eeba42a3e4cb
2015/05/15 15:18:28 rafthttp: request ignored due to cluster ID mismatch got ce6de360c0134f3 want 7e27652122e8b2ae
```

Output from cluster:
```
2015/05/15 13:18:30 INFO: raft: 69e5eeba42a3e4cb [logterm: 6, index: 6214] sent vote request to e617c34902234818 at term 8
2015/05/15 13:18:30 rafthttp: roundtripping error: error roundtripping to http://10.128.1.203:2380/raft/stream/message/69e5eeba42a3e4cb: dial tcp 10.128.1.203:2380: connection refused
2015/05/15 13:18:30 rafthttp: roundtripping error: error roundtripping to http://10.128.1.203:2380/raft/stream/msgapp/69e5eeba42a3e4cb: dial tcp 10.128.1.203:2380: connection refused
2015/05/15 13:18:30 rafthttp: roundtripping error: error roundtripping to http://10.128.1.153:2380/raft/stream/message/69e5eeba42a3e4cb: dial tcp 10.128.1.153:2380: connection refused
2015/05/15 13:18:30 rafthttp: roundtripping error: error roundtripping to http://10.128.1.153:2380/raft/stream/msgapp/69e5eeba42a3e4cb: dial tcp 10.128.1.153:2380: connection refused
2015/05/15 13:18:30 etcdserver: conflicting cluster ID with the target cluster (7e27652122e8b2ae != ce6de360c0134f3)
2015/05/15 13:18:30 etcdserver: the data-dir used by this member must be removed.
```

I expected a failure to join, but not killed cluster. The first time I did it, was by accident.

**Comment by xiang90** [comment]
@mariusgrigaitis We can simply ignore the request from both side and keep on printing out warnings.
Is this what you would expect?

**Comment by mariusgrigaitis** [comment]
@xiang90 I expect cluster to be alive and crash-proof. So probably yes - warning in logs that someone is trying to join with different id.

**Comment by xiang90** [comment]
Fixed by #2871
